neural_modelling/src/neuron/population_table/population_table_binary_search_impl.c

```diff
@@ -54,16 +57,32 @@ static inline uint32_t get_address(address_and_row_length entry) {
 }
 
 static inline uint32_t get_row_length(address_and_row_length entry) {
-    return entry & 0xFF;
+    // Row lengths are stored offset by 1, to allow 1-256 length rows
+    return (entry & 0xFF) + 1;
 }
 
 static inline uint32_t is_single(address_and_row_length entry) {
     return entry & 0x80000000;
 }
 
+static inline uint32_t get_n_neurons(master_population_table_entry entry) {
+    return entry.n_neurons_and_mask_shift >> 5;
```

Comment:
magic number

what happens if neurons is from a retina????

Reply:
Fixed.  Retina will work because it is a single entry anyway, but I have added checks elsewhere to avoid issues.